HList.cpp

```diff
@@ -0,0 +1,332 @@
+		curent_ref = curent_ref -> next;
+
+	}
+
+}
+void HList::addElement(int new_data, int position) {
+
+	if (position > n || position < 0) {
+		std::cerr << "Position denied." << std::endl;
+		return;		
+	}
+
+	Node *new_node = new Node{0,NULL,NULL};
+	new_node -> data = new_data;
+
+	Node *curent_ref = head_ref;
+
+	if (position == 0) {
+		if (head_ref == NULL) {
+			head_ref = new_node;
+			tail = new_node;
+		}
+		else {
+			head_ref -> prev = new_node;
+			new_node -> next = head_ref;
+			head_ref = new_node;
+		}
+
+	}
+	else if (position == n) {
+		new_node -> prev = tail;
+		tail -> next = new_node;
+		tail = new_node;
+		new_node -> next = NULL;
+	}
+	else {
+
+		curent_ref = thru(position);
+		new_node -> next = curent_ref;
+		new_node -> prev = curent_ref -> prev;
+		curent_ref -> prev -> next = new_node;
+		curent_ref -> prev = new_node;
+		
+	}
+
+	n++;
+}
+
+
+HList::HList():		// THE default CONSTRUCTOR //
+	head_ref(NULL), tail(NULL), n(0)
+{
+
+}
+
+HList::~HList() {   // THE DESTRUCTOR //
+
+	Node *curent_ref = head_ref;
+
+	while (curent_ref != NULL){
+
+		curent_ref = curent_ref -> next;
+
+		if(curent_ref && curent_ref -> prev){
+			delete curent_ref -> prev;
+		}
+
+	}
+	if(tail) {
+		delete tail;
+	}
+}
+
+void HList::addFront(int new_data) {
+
+	// Alocam un nou nod de date.
+
+	Node *new_node = new Node;
+
+	// Ii dam input datele pe care dorim sa le stocam
+
+	new_node -> data = new_data;
+
+	// Primul nod, nu are un antecetend, dar are urmatorul item ,
+	// itemul 2(Fostul item 1) (Practic toate itemele devin item n+1.
+
+	new_node -> next = (head_ref);
+	new_node -> prev = NULL;
+
+	if (head_ref != NULL) {
+		(head_ref) -> prev = new_node;
+	} 
+	else {
+		tail = new_node;
+	}
+
+	(head_ref) = new_node;
+
+	n++;
+	// std::cout << p -> next << std::endl;
+
+}
+
+void HList::addBack(int new_data) {
+
+	Node *new_node = new Node{0,NULL,NULL};
+	new_node -> data = new_data;
+
+	if(tail != NULL) {
+		tail -> next = new_node;
+		new_node -> prev = tail;
+		tail = new_node;
+}
+	else {
+		head_ref = new_node;
+		tail = new_node;
+	}
+
+}
+
+void HList::removeBack() {
+
+	if(tail == NULL) {
+		std::cerr << "Tail is NULL" << std::endl;
+		return;
+	}
+	Node *tail_del = tail;
+	tail = tail -> prev;
+	tail_del -> prev -> next = NULL;
+	delete tail_del;
+
+}
+
+void HList::removeFront() {
+
+	if(head_ref == NULL) {
+		std::cerr << "Head is NULL" << std::endl;
+		return;
+	}
+	else if(head_ref == tail) {
+		delete head_ref;
+		delete tail;
+	}
+	else {
+	Node *head_del;
+	head_ref = head_ref -> next;
+	head_del -> next -> prev = NULL;
+	delete head_del;
+	}
+}
+
+int HList::searchFirstOccurence(int data) {
+
+	Node *curent_ref;
+	curent_ref = head_ref;
+	int curent_pos = 0;
+
+	if(curent_ref == NULL) {
+		std::cerr << "You can't do that while the list is empty." << std::endl;
+		return 0;
+	}
+
+	while(curent_ref -> data != data) {
```

Comment:
super, dar aici tot iti trebuie sa verifici daca curent_ref e NULL pentru ca tot avansezi
```suggestion
	while(curent_ref && curent_ref -> data != data) {
```